intellij/src/saros/intellij/runtime/EDTExecutor.java

```diff
@@ -0,0 +1,130 @@
+   *
+   * @param computation the computation to run
+   * @param modalityState the modality state to use
+   * @param <T> the type of the result of the computation
+   * @param <E> the type of the exception that might be thrown by the computation
+   * @return returns the result of the computation
+   * @throws E any exception that occurs while executing the computation
+   * @see Application#invokeAndWait(Runnable,ModalityState)
+   */
+  @SuppressWarnings("unchecked")
+  public static <T, E extends Throwable> T invokeAndWait(
+      @NotNull ThrowableComputable<T, E> computation, @NotNull ModalityState modalityState)
+      throws E {
+
+    AtomicReference<T> result = new AtomicReference<>();
+    AtomicReference<Throwable> throwable = new AtomicReference<>();
+
+    application.invokeAndWait(
+        () -> {
+          try {
+            result.set(computation.compute());
+
+          } catch (Throwable t) {
+            throwable.set(t);
+          }
+        },
+        modalityState);
+
+    Throwable t = throwable.get();
+
+    if (t == null) return result.get();
+
+    if (t instanceof Error) throw (Error) t;
+
+    if (t instanceof RuntimeException) throw (RuntimeException) t;
+
+    throw (E) t;
+  }
+
+  /**
+   * Executes the given computation synchronously on the EDT using the given modality state and
+   * returns the result of the computation.
+   *
+   * @param computation the computation to run
+   * @param modalityState the modality state to use
+   * @param <T> the type of the result of the computation
+   * @return the result of the computation
+   * @see Application#invokeAndWait(Runnable, ModalityState)
+   */
+  public static <T> T invokeAndWait(
+      @NotNull Computable<T> computation, @NotNull ModalityState modalityState) {
+
+    AtomicReference<T> result = new AtomicReference<>();
+
+    application.invokeAndWait(() -> result.set(computation.compute()), modalityState);
+
+    return result.get();
+  }
+
+  /**
+   * Calls {@link #invokeAndWait(Computable, ModalityState)} with {@link
+   * ModalityState#defaultModalityState()}.
+   */
+  public static <T> T invokeAndWait(@NotNull Computable<T> computation) {
+    return invokeAndWait(computation, ModalityState.defaultModalityState());
+  }
+
+  /**
+   * Executes the given runnable synchronously on the EDT using the given modality state.
+   *
+   * @param runnable the runnable to execute
+   * @param modalityState the modality state to use
+   * @see Application#invokeAndWait(Runnable, ModalityState)
+   */
+  public static void invokeAndWait(
+      @NotNull Runnable runnable, @NotNull ModalityState modalityState) {
+
+    application.invokeAndWait(runnable, modalityState);
+  }
+
+  /**
+   * Calls {@link #invokeAndWait(Runnable, ModalityState)} with {@link
+   * ModalityState#defaultModalityState()}.
+   */
+  public static void invokeAndWait(@NotNull Runnable runnable) {
+    invokeAndWait(runnable, ModalityState.defaultModalityState());
```

Comment:
There is already an [invokeAndWait method that uses the defaultModalityState](https://github.com/JetBrains/intellij-community/blob/5ce01ec91f0eae2b44a0f6e712eb219863967ce9/platform/core-api/src/com/intellij/openapi/application/Application.java#L386)
Same for `invokeLater`

Reply:
And, as far as I can tell, [they do exactly the same](https://github.com/JetBrains/intellij-community/blob/5ce01ec91f0eae2b44a0f6e712eb219863967ce9/platform/platform-impl/src/com/intellij/openapi/application/impl/ApplicationImpl.java#L485). But, with these being convenience methods, I thought it would be better to specify this assumption in our code and not rely on theirs.